### PR TITLE
Pxapi2/feature/pbi152452 codelist endpoint

### PR DESCRIPTION
**Pull request by MikaelNordberg**
Implemented Codelist endpoint.
Danish PX-file with aggregations added.
Links only in current language.
Moved logic for time interval to CnmmDataSource.
RANGE selection expression handles if values in wrong order.
Changed log level.

### CODE REVIEW

File: PxWeb/Code/Api2/DataSource/Cnmm/CnmmDataSource.cs [hunk 1]
```diff
@@ -90,12 +96,83 @@ public CnmmDataSource(ICnmmConfigurationService cnmmConfigurationService, IItemS
...
+        {
+            Codelist? codelist = null;
+            var cnmmOptions = _cnmmConfigurationService.GetConfiguration();
+
+            if (string.IsNullOrEmpty(id))
+            {
+                return codelist;
+            }
+
+            if (id.StartsWith("agg_", System.StringComparison.InvariantCultureIgnoreCase))
```
Comment (likp) [inline PxWeb/Code/Api2/DataSource/Cnmm/CnmmDataSource.cs:109]: Maybe splitt this into smaller private methods that you can call for readability.

File: PxWeb/Code/Api2/DataSource/PxFile/PxFileDataSource.cs [hunk 2]
```diff
@@ -83,6 +87,27 @@ public PxFileDataSource(IPxFileConfigurationService pxFileConfigurationService,
...
+        public Codelist? GetCodelist(string id, string language)
+        {
+            Codelist? codelist = null;
+            
+            if (string.IsNullOrEmpty(id))
+            {
+                return codelist;
+            }
+
+            id = id.Replace("agg_", "");
```
Comment (likp) [inline PxWeb/Code/Api2/DataSource/PxFile/PxFileDataSource.cs:99]: We sould olny remove agg_ from the beginning of the id.

**Review by likp — COMMENTED** [review]
See comments, oterwise it looks great!

File: PxWeb/Code/Api2/DataSource/PxFile/PxFileDataSource.cs (hunk 2 above)
Comment (MikaelNordberg) [reply PxWeb/Code/Api2/DataSource/PxFile/PxFileDataSource.cs]: Now uses substring instead

File: PxWeb/Code/Api2/DataSource/Cnmm/CnmmDataSource.cs (hunk 1 above)
Comment (MikaelNordberg) [reply PxWeb/Code/Api2/DataSource/Cnmm/CnmmDataSource.cs:108]: Made private methods to get CNMM Groupings and Valuesets

**Review by likp — APPROVED** [review]

**Pull request merged**